src/test/java/com/puppycrawl/tools/checkstyle/bdd/InputConfiguration.java

```diff
@@ -0,0 +1,141 @@
+
+    /** The check name. */
+    private final String checkName;
+
+    /** Map of default properties. */
+    private final Map<String, String> defaultProperties;
+
+    /** Map of non default properties. */
+    private final Map<String, String> nonDefaultProperties;
+
+    /** List of violations. */
+    private final List<String> violations;
+
+    private InputConfiguration(String checkName,
+                               Map<String, String> defaultProperties,
+                               Map<String, String> nonDefaultProperties,
+                               List<String> violations) {
+        this.checkName = checkName;
+        this.defaultProperties = defaultProperties;
+        this.nonDefaultProperties = nonDefaultProperties;
+        this.violations = violations;
+    }
+
+    public String getCheckName() {
+        return checkName;
+    }
+
+    public Map<String, String> getAllProperties() {
+        final Map<String, String> properties = new HashMap<>();
+        properties.putAll(defaultProperties);
+        properties.putAll(nonDefaultProperties);
+        return Collections.unmodifiableMap(properties);
+    }
+
+    public Map<String, String> getDefaultProperties() {
+        return Collections.unmodifiableMap(defaultProperties);
+    }
+
+    public Map<String, String> getNonDefaultProperties() {
+        return Collections.unmodifiableMap(nonDefaultProperties);
+    }
+
+    public List<String> getViolations() {
+        return Collections.unmodifiableList(violations);
+    }
+
+    public DefaultConfiguration createConfiguration() {
+        return new DefaultConfiguration(checkName);
+    }
+
+    public static final class Builder {
+
+        private final Map<String, String> defaultProperties = new HashMap<>();
+
+        private final Map<String, String> nonDefaultProperties = new HashMap<>();
+
+        private final List<String> violations = new ArrayList<>();
+
+        private String checkName;
+
+        public Builder withCheckName(String inputCheckName) {
+            checkName = inputCheckName;
+            return this;
+        }
+
+        public Builder withDefaultProperties(Map<String, String> defaultPropertiesMap) {
+            if (defaultPropertiesMap != null) {
+                defaultProperties.putAll(defaultPropertiesMap);
+            }
+
+            return this;
+        }
+
+        public Builder withNonDefaultProperties(Map<String, String> nonDefaultPropertiesMap) {
+            if (nonDefaultPropertiesMap != null) {
+                nonDefaultProperties.putAll(nonDefaultPropertiesMap);
+            }
+
+            return this;
+        }
+
+        public Builder withViolationList(List<String> violationList) {
+            if (violationList != null) {
+                violations.addAll(violationList);
+            }
+
+            return this;
+        }
+
+        public void setCheckName(String checkName) {
+            this.checkName = checkName;
+        }
```

Comment:
return `this`, same for `addDefaultProperty`, `addNonDefaultProperty ` and `addViolation `

Reply:
Why? Won't we be adding these one by one?

Reply:
sure, lets skip it